test/serializer-test.ts

```diff
@@ -263,4 +268,311 @@ describe("serializer", function(): void {
+    const account = "r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ";
+    const publicKey =
+      "031D68BC1A142E6766B2BDFB006CCFE135EF2E0E2E94ABB5CF5C9AB6104776FBAE";
+
+    const paymentAmount = new XRPDropsAmount();
+    paymentAmount.setDrops(value);
+
+    const currencyAmount = new CurrencyAmount();
+    currencyAmount.setXrpAmount(paymentAmount);
+
+    const destinationAddress = new AccountAddress();
+    destinationAddress.setAddress(destination)
+
+    const payment = new Payment();
+    payment.setDestination(destinationAddress);
+    payment.setAmount(currencyAmount);
+
+    const transactionFee = new XRPDropsAmount();
+    transactionFee.setDrops(fee);
+
+    const sender = new AccountAddress();
+    sender.setAddress(account);
+
+    const transaction = new Transaction();
+    transaction.setAccount(sender);
+    transaction.setFee(transactionFee);
+    transaction.setSequence(sequence);
+    transaction.setPayment(payment);
+    transaction.setSigningPublicKey(Utils.toBytes(publicKey));
+    transaction.setLastLedgerSequence(lastLedgerSequence);
+
+    // WHEN the transaction is serialized to JSON.
+    const serialized = Serializer.transactionToJSON(transaction);
+
+    // THEN the result is as expected.
+    const expectedJSON = {
+      Account: account,
+      Amount: value.toString(),
+      Destination: destination,
+      Fee: fee.toString(),
+      LastLedgerSequence: lastLedgerSequence,
+      Sequence: sequence,
+      TransactionType: "Payment",
+      SigningPubKey: publicKey
+    };
+    assert.deepEqual(serialized, expectedJSON);
+  });
+
+  it("serializes a payment in XRP from an X-Address with no tag", function(): void {
+    // GIVEN a transaction which represents a payment denominated in XRP.
+    const value = 1000;
+    const destination = "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh";
+    const fee = 10;
+    const lastLedgerSequence = 20;
+    const sequence = 1;
+    const account = "XVPcpSm47b1CZkf5AkKM9a84dQHe3m4sBhsrA4XtnBECTAc";
+    const publicKey =
+      "031D68BC1A142E6766B2BDFB006CCFE135EF2E0E2E94ABB5CF5C9AB6104776FBAE";
+
+    const paymentAmount = new XRPDropsAmount();
+    paymentAmount.setDrops(value);
+
+    const destinationAddress = new AccountAddress();
+    destinationAddress.setAddress(destination)
+
+    const currencyAmount = new CurrencyAmount();
+    currencyAmount.setXrpAmount(paymentAmount);
+
+    const payment = new Payment();
+    payment.setDestination(destinationAddress);
+    payment.setAmount(currencyAmount);
+
+    const transactionFee = new XRPDropsAmount();
+    transactionFee.setDrops(fee);
+
+    const sender = new AccountAddress();
+    sender.setAddress(account);
+
+    const transaction = new Transaction();
+    transaction.setAccount(sender);
+    transaction.setFee(transactionFee);
+    transaction.setSequence(sequence);
+    transaction.setPayment(payment);
+    transaction.setSigningPublicKey(Utils.toBytes(publicKey));
+    transaction.setLastLedgerSequence(lastLedgerSequence);
```

Comment:
Do you want to move all these steps into a helper method that builds up a transaction? Seems like this structure is duplicated across tests.

Reply:
Done. I also tried to pull out certain fields to constants as well. This file is getting unwieldy at the moment. Fortunately, we should be able to deprecate and remove the `legacy` code soon which should make things a bit better.